hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java

```diff
@@ -289,10 +289,13 @@ private synchronized void writeCurrentBufferToService() throws IOException {
     final Future<Void> job = completionService.submit(new Callable<Void>() {
       @Override
       public Void call() throws Exception {
-        client.append(path, offset, bytes, 0,
-            bytesLength);
-        byteBufferPool.putBuffer(ByteBuffer.wrap(bytes));
-        return null;
+        try (AbfsPerfInfo tracker = new AbfsPerfInfo(client.getAbfsPerfTracker(), "writeCurrentBufferToService", "append")) {
```

Comment:
This will likely tirgger a checkstyle >80 chars.
Probably extract the perf tracker.
In addition, what about having constants for the strings?

Reply:
the abfs code hasn't been that strict about 80 chars, but it is nice to have a length which makes it easy for side-by-side diff reviews. Here I'd put a newline after the = sign

Reply:
I have split the line.
The constant strings here are actually the names of the containing method and the about-to-be-called method. Moreover, we need these strings at only one place: this line where "writeCurrentBufferToService" is calling "append". Don't you think that pulling all such method names out into a long list of constants will bloat up the code?

Reply:
@goiri Did you give thought to the suggestion in the last comment?

Reply:
@goiri based on our discussions on email thread, I am resolving this conversation.